TestProjects/iOS-Support/XamarinSidebar/MvvmCross.iOS.Support.Core/App.cs

```diff
@@ -13,7 +15,7 @@ public override void Initialize()
                 .AsInterfaces()
                 .RegisterAsLazySingleton();
 
-            RegisterAppStart(new AppStart());
+            RegisterAppStart(new AppStart(Mvx.Resolve<IMvxNavigationService>()));
```

Comment:
Turns out you'll need to resolve it somewhere since the `App` class is constructed in the platform specific `Setup` class and `Setup` is constructed in the `AppDelegate` class (in case of iOS). 

Resolving is in `AppDelegate` is to early (since the service is not registered yet at this point). The other option is to resolve it in the `Setup` class but this would mean you would potentially duplicate code (of course not here since it is a iOS only example, but for other projects this might be the case). So I think the best place would be to resolve it where it currently is. Of course I am open for suggestions in case I am missing something.